src/main/java/hudson/plugins/ec2/EC2Cloud.java

```diff
@@ -475,6 +484,15 @@ private int countCurrentEC2Slaves(SlaveTemplate template) throws AmazonClientExc
         return n;
     }
 
+    private boolean isEc2ProvisionedJenkinsSlave(List<Tag> tags, String serverUrl) {
```

Comment:
imho, it would be better to use `getCustomInstanceTags` here and compare list of tags from instance with expected tags.